src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateTargetLocator.java

```diff
@@ -7,24 +7,49 @@
 import io.stargate.sgv2.jsonapi.exception.JsonApiException;
 import java.util.regex.Pattern;
 
-/** Factory for {@link UpdateTarget} instances. */
+/** Helper class for resolving "dotted paths" into {@link UpdateTarget} instances. */
 public class UpdateTargetLocator {
   private static final Pattern DOT = Pattern.compile(Pattern.quote("."));
 
   private static final Pattern INDEX_SEGMENT = Pattern.compile("0|[1-9][0-9]*");
 
+  private final String dotPath;
+
+  private final String[] segments;
+
+  private UpdateTargetLocator(String dotPath, String[] segments) {
+    this.dotPath = dotPath;
+    this.segments = segments;
+  }
+
+  public String path() {
+    return dotPath;
+  }
+
   /**
-   * Method that will create target instance for given path through given document; if no such path
-   * exists, will not attempt to create path (nor report any problems).
+   * Factory method for constructing path; also does minimal verification of path: currently only
+   * verification is to ensure there are no empty segments.
+   *
+   * @param dotPath Path that uses dot-notation
+   * @return Locator instance
+   * @throws JsonApiException if dotPath invalid (empty path segment(s))
+   */
+  public static UpdateTargetLocator forPath(String dotPath) throws JsonApiException {
+    return new UpdateTargetLocator(dotPath, splitAndVerify(dotPath));
```

Comment:
May not be for this PR, should we create UpdateTargetLocator.forPath() inside class implementing ActionWithPath? I see the splitAndVerify() will be done repeatedly for the same path in case of updateMany.

Reply:
@maheshrajamani Yes, exactly, I will need to further refactor things to move construction earlier. Will work on that in the follow-up PR.